libkineto/src/RoctracerLogger.cpp

```diff
@@ -24,6 +26,21 @@ using namespace std::chrono;
 
 constexpr size_t kBufSize(2 * 1024 * 1024);
 
+class Flush
+{
+public:
+  std::atomic<bool> doFlush_ {false};
+  std::mutex mutex_;
+  std::atomic<uint64_t> maxCorrelationId_;
+  uint64_t correlationId_ {0};
+  void reportCorrelation(const uint64_t &cid) {
+    uint64_t prev = maxCorrelationId_;
+    while (prev < cid && maxCorrelationId_.compare_exchange_weak(prev, cid))
```

Comment:
```suggestion
    while (prev < cid && !maxCorrelationId_.compare_exchange_weak(prev, cid))
```
Should this need "!"?  compare_exchange_weak returns true if the underlying atomic value was successfully changed.

Reply:
Opps.  Good catch.  Amazingly it (appeared) to work fine as it was, or else I did an accidental edit.  compare_exchange_weak is allowed to return false spuriously (and doesn't update prev), that may be what let it break out of the loop.  Crazy.